antsibull/schemas/base.py

```diff
@@ -261,6 +279,18 @@ def merge_typo_names(cls, values):
 
         return values
 
+    @p.root_validator
+    def require_version_xor_date(cls, values):
+        """Make sure either removed_in or removed_at_date are specified, but not both."""
+        has_removed_in = values.get('removed_in', _SENTINEL) is _SENTINEL
+        has_removed_at_date = values.get('removed_at_date', _SENTINEL) is _SENTINEL
+        if not has_removed_in and not has_removed_at_date:
+            raise ValueError('One of `removed_at_date` and `removed_in` must be specified')
+        if has_removed_in and has_removed_at_date:
+            raise ValueError('Not both of `removed_at_date` and `removed_in` can be specified')
+
+        return values
+
```

Comment:
Since we would eventually like this mutual exclusion to show up in the json schema, I think we should open an issue and add a comment here to point at it.

I think this is the pydantic feature which will allow us to add an xor to the json schema but it will take some effort so it's okay with me if we just point it out and don't implement it in this PR:

* https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization

Whoever does implement it will also need to know a little about json-schema.  I think this is the heart of what we'll need to output:
* http://json-schema.org/understanding-json-schema/reference/combining.html#oneof

Reply:
#104